New version

samary opened the pull request:
This is a new version of the Quattor Dashboard,

This project is a fork from : https://github.com/amazerfrazer/aquilonconsole
# Available pages
- Profiles
  - List of hosts with boot status and AII actions
  - Profile reader
- Overview
  - List of hosts with related profiles fields
- SVN Logs
  - Last SVN logs
- Statistics
  - Statistics of hosts
- Aquilon (NOT TESTED)
  - AZ
  - Hosts
  - Personalities
# Screenshots

![selection_011](https://cloud.githubusercontent.com/assets/5583798/6846267/0df2d3aa-d3bd-11e4-8b7e-e4d7fd915859.png)

![selection_012](https://cloud.githubusercontent.com/assets/5583798/6846266/0a4601be-d3bd-11e4-823f-472f4489040c.png)

![selection_013](https://cloud.githubusercontent.com/assets/5583798/6846264/068001ba-d3bd-11e4-9886-093eb98a0d88.png)

Commit:
New Version

Commit:
Remove old files

jrha commented:
Nice! Will review later.

Commit:
Add Apache2 License

Commit:
Create .gitignore

Ingoring :
- IDE settings
- Remote cgi test
- Generated CSS

stdweird commented on vhost-dashboard.conf:
```diff
@@ -0,0 +1,21 @@
+NameVirtualHost MY_HOST:80
```
how is this used? there is some templating?

samary replied:
You have the explanation of this in the README
FYI: I use the NameVirtualHost clause because I run multiple website on the same webserver.

stdweird commented on templates/svnlogs.html:
```diff
@@ -0,0 +1,41 @@
+        <tbody></tbody>
+    </table>
+</div>
+{% endblock %}
```
@samary a general remark for the whole PR: end with a newline

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+# ${developer-info
+# ${author-info}
+# ${build-info}
+
```
add 

``` perl
use strict;
use warnings;
```

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+
+use CGI;
+
+my $repo = "file:///var/www/svn/scdb";
```
should be read from config file

samary replied:
Do you have a place to suggest where to put it ?
AFAIU, there is no SVN repository definition in quattor itself

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+
+my $repo = "file:///var/www/svn/scdb";
+
+my $q = new CGI;
```
use longer variable name

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+
+my $q = new CGI;
+
+my $limit = $q->param('limit') ? $q->param('limit') : '10';
```
perl has no types, use `10` instead of `'10'`

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+
+my $limit = $q->param('limit') ? $q->param('limit') : '10';
+
+$limit =~ /(\d+|all)/;
```
use strict regexp `m/^(\d+|all)$/`

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+my $limit = $q->param('limit') ? $q->param('limit') : '10';
+
+$limit =~ /(\d+|all)/;
+$limit = $1;
```
this needs error handling like

``` perl
if ($limit =~ m/^(\d+|all)$/) {
    $limit = $1;
} else {
   exit with clean html error
}
```

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,30 @@
+my $output;
+
+if($limit eq 'all') {
+  $output = `/usr/bin/sudo /usr/bin/svn log --xml $repo`;
```
no backticks. build up a command array and use `CAF::Process`
e.g.

``` perl
my @command = qw(/usr/bin/sudo /usr/bin/svn log --xml);
push(@command, '-l', $limit, '-r', 'HEAD:1') if ($limit ne 'all');
push(@command, $repo);

my $p = CAF::Process(\@command);
$ouput = $p->output();
# do something with $? exitcode in case of failure
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+# ${developer-info
+# ${author-info}
+# ${build-info}
+
```
add 

``` perl
use strict;
use warnigns;
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+my $boothd = "default";
+
+# Quattor profiles
+my $web_root = "/var/www/html";
```
read from config file

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+# Quattor profiles
+my $web_root = "/var/www/html";
+my $profiles_path = "profiles";
```
read from config file

samary replied:
I don't find where this is or should be declared, in aii.conf (it uses `cdb_url` to fetch profiles) ?
Adding fields in `aii.conf` ?

If I use the `cdb_url` parameter, I have to switch to an 'over network' version of this code (read files locally vs fetch via url).

stdweird replied:
as this is an `aii-shellfe` box, you can use indeed the settings from `/etc/aii/aii-shellfe.conf` as you do below, and read the values from there, e.g. `cdburl`
we have `cdburl=file:///path/to/profiles` in our AII servers

samary replied:
@stdweird I changed to use this parameter, this force me to allow only file:// protocol. If I want to handle http://, I need to use LWP to get profiles. I suggest to leave this as is and I'll figure a way to work with the AII framework itself for cleaner handling. What do you think ?

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+my $profiles_dir = $web_root.'/'.$profiles_path;
+
+# Read AII configuration
+$cfg = new Config::Simple();
```
maybe use `Config::Tiny` if possible, since that is standard part of quattor (but i'm not sure it can handle the format)

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+my $pxelinux_dir = $cfg->param('nbpdir');
+
+# Profile prefix
+my $profile_prefix = $cfg->param('profile_prefix') eq 0 ? '' : $cfg->param('profile_prefix');
```
rework the logic?

``` perl
my $profile_prefix = $cfg->param('profile_prefix') || '';
```

samary replied:
I have in my aii-shellfe.conf : profile_prefix = 0. I don't know how this value is threated by aii itself. Is it required by aii and what type is used (I don't understand how 0 is known as empty for AII) ?

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+# Profiles .json available
+my (@profiles);
+
+################################################################
```
remove this `#######################`

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+################################################################
+sub GetHexAddr {
+################################################################
+# Get the hostname or an IP address and return the IP address in
```
put the comments before the method

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+my (@profiles);
+
+################################################################
+sub GetHexAddr {
```
move the `{` to the next line

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+# hex (as required by pxelinux)
+
+  # The 4th field is an array of the IP address of this node
+  my @all_address;
```
use

``` perl
my @all_address =(gethostbyname($_[0]))[4];
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+# Get the hostname or an IP address and return the IP address in
+# hex (as required by pxelinux)
+
+  # The 4th field is an array of the IP address of this node
```
perl code has 4 space indent

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  # The 4th field is an array of the IP address of this node
+  my @all_address;
+  @all_address =(gethostbyname($_[0]))[4];
+  if ($#all_address < 0) { # The array is empty
```
use

``` perl
return if (@all_address);
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  # We unpack the IP address
+  my @tmp_address = unpack('C4',$all_address[0]);
+  my @result;
+  $result[0]=sprintf ("%02X%02X%02X%02X",$tmp_address[0], $tmp_address[1],
```
space around the `=`

stdweird replied:
use `push` and pass `@tmp_address` as arg to sprintf (it's all the same in perl)

```
push(@result, sprintf ("%02X%02X%02X%02X", @tmp_address));
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  my @result;
+  $result[0]=sprintf ("%02X%02X%02X%02X",$tmp_address[0], $tmp_address[1],
+                                         $tmp_address[2], $tmp_address[3]);
+  $result[1]=sprintf ("%u.%u.%u.%u",$tmp_address[0], $tmp_address[1],
```
same remark

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+}
+
+
+#########################################################################
```
remove

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+sub Initialize {
+#########################################################################
+
+  @profiles=();
```
`my`, and space around `=`

samary replied:
`@profiles` declared in the upper scope

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+  @profiles=();
+
+  opendir(DIR,$profiles_dir);
```
handle error like `opendir() || die "failed to opendir $profiles_dir: $!"`

stdweird replied:
where does `profiles_dir` come from? should be passed as argument

samary replied:
`profiles_dir´ comes from the upper scope

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  opendir(DIR,$profiles_dir);
+
+  # find all profiles in directory
+  push @profiles,map { s/\.json$//; s/^$profile_prefix//; $_ .=''; } sort(grep(/\.json$/, readdir(DIR)));
```
support `json.gz`. 
what is the map code supposed to do?

samary replied:
OK for json.gz.
The map is to extract the hostname from the profile filename. (remove .json and $profile_prefix)

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+  closedir(DIR);
+  # Load the configurations list
+  opendir(DIR, $pxelinux_dir);
```
add error handling

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  closedir(DIR);
+  # Load the configurations list
+  opendir(DIR, $pxelinux_dir);
+  @cfg = sort(grep(/(\.cfg$)|(default)/, readdir(DIR)));
```
what is this supposed to do?

stdweird replied:
also `my @cfg`

samary replied:
`@cfg` declared above.
This gather the list of pxelinux cfg file.

`default` is for : 
 Following pxelinux convention it must be called default so if the <IpAddressInHex> link is missing the node will boot from the hard disk

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  opendir(DIR, $pxelinux_dir);
+  @cfg = sort(grep(/(\.cfg$)|(default)/, readdir(DIR)));
+  closedir(DIR);
+}
```
return something?

samary replied:
Nope, just initialize required arrays (hostnames and pxeboot configs).
Need to rework this logic.

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+  my (@all, $k, $json);
+
+  for $k (@profiles) {
```
use meaningful variable name 

``` perl
foreach my $profile (@profiles) {
```

where does `@profiles` come from? should be passed as argument

samary replied:
`@profiles` is declared above and populated by the `Initialize` function

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+                                         $tmp_address[2], $tmp_address[3]);
+  $result[1]=sprintf ("%u.%u.%u.%u",$tmp_address[0], $tmp_address[1],
+                                    $tmp_address[2], $tmp_address[3]);
+  return @result;
```
you can also do

``` perl
return sprintf ("%02X%02X%02X%02X", @tmp_address), sprintf ("%u.%u.%u.%u",@tmp_address);
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+    my $hostname = $k;
+
+    my ($hexaddr,$dotaddr) = GetHexAddr($hostname);
```
space after `,`

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+    my ($hexaddr,$dotaddr) = GetHexAddr($hostname);
+
+    my $link = readlink("$pxelinux_dir/$hexaddr");
```
test for existence with 

``` perl
my $link = "$pxelinux_dir/$hexaddr"; 
if (! -f $link) {
   error
   next;
}
...
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+    my $link = readlink("$pxelinux_dir/$hexaddr");
+    my $existing_cfg = $link ? $link : "";
+
+    push @all ,{'hostname' => $hostname, 'hexaddr' => $hexaddr, 'dotaddr' => $dotaddr, 'bootcfg' => $existing_cfg };
```
each key of the hashref on a new line

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+sub GetProfile {
+#########################################################################
+
+  if ($_[0] =~ /^(.*)$/) {
```
don't use `$_` like this. it has subtle sideeffects 
use something like 

``` perl
my ($hostname) = @_;
```

also use a better regexp to untaint (`m/^([\w-.]+)$/`)

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+      my $json =  JSON::XS->new->pretty->allow_nonref->encode($profile);
+
+      print "Content-type: application/json\n\n$json";
+    }
```
handle the else block with a nice error message?

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  if ($_[0] =~ /^(.*)$/) {
+    $_[0] = $1;
+    my $hostname = $_[0];
+    my $file = "$profiles_dir/$profile_prefix$hostname.json";
```
gzip support

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+      my $profile = JSON::XS->new->decode($json_text);
+      
+      #Remove sensitive sections
+      $profile->{'software'} = ();
```
this removes quite a lot, and if you want to hide sensitive data, this is probably not enough anyway.

samary replied:
True, any suggestion is welcome !
Is there any exhaustive list of sensitive data in the profiles (path list) ?

jrha replied:
It could vary wildly by site, but you could start with the root password hash.

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  my $result = '';
+  print "Content-type: text/plain\n\n";
+
+  $_[0] =~ /^(configure|install|reinstall|boot)$/;
```
add remove

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+
+  my $result = '';
+  print "Content-type: text/plain\n\n";
+
```
use 

``` perl
my ($action, $hostname) = @_;
```

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  print "Content-type: text/plain\n\n";
+
+  $_[0] =~ /^(configure|install|reinstall|boot)$/;
+  $_[0] = $1;
```
handle the non-matching case with error message

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+  my $hostname = $_[1];
+
+  if ($action eq 'reinstall') {
+    my $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--remove", $hostname);
```
use `CAF::process`, check exit code and i think you can do `--remove --configure --install` in one go

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+    my $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--remove", $hostname);
+    $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--configure", $hostname);
+    $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--install", $hostname);
+    print "Failed to reinstall $node using aii-shellfe<br>" if ($ok != 0);
```
use `$?` for the exitcode with `CAF::Process`

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+    if (system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--$action", $hostname) != 0) {
+        print "Failed to $action $node using aii-shellfe<br>";
+      }
+  }
```
but try to build the command array and make one call to `CAF::Process`

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,335 @@
+#########################################################################
+
+  my (%all, %result, $k, $i, $json, $hostname, $value);
+
```
@samary i'm going to stop here. some of the above comments also apply below, i'll continue when you made your first round of cleanup :smile: 
but i saw the demo, very nice work!

samary replied:
@stdweird Thanks for this review. I will work on your comments.

Commit:
add use strict and warnings

Commit:
add longer variable names

Commit:
untyping

Commit:
stricter regex

Commit:
add use strict and warnings

Commit:
[FIX] warnings and errors due to use strict and warnings

Commit:
Use perl Config Tiny

Commit:
use pod format for functions documentation

Commit:
move opening brace to newline for functions

Commit:
cleanup GetHexAddr code

Commit:
cleanup Initialize code

Commit:
cleanup GetHexAddr code + typos

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+1- Install required software & scripts
+
+* On the web server
+    - `yum install httpd mod_wsgi mod_proxy git python-cherrypy python-jinja2`
```
add `-y`

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+
+* On the web server
+    - `yum install httpd mod_wsgi mod_proxy git python-cherrypy python-jinja2`
+    - `git clone https://github.com/IIHE/dashboard.git`
```
what is located here that can't merged in this repo?

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+* On the web server
+    - `yum install httpd mod_wsgi mod_proxy git python-cherrypy python-jinja2`
+    - `git clone https://github.com/IIHE/dashboard.git`
+    - Copy the vhost-dashboard.conf to your Apache config directory (eg. /etc/http/conf.d/) and replace the following:
```
copy from where? (if it's packaged in the rpm, add the location)

stdweird replied:
also add backticks around filenames and paths

samary replied:
I don't know if this repo will ever be packaged as RPM.
You copy the file from the clone you made in the previous step

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+    - `yum install httpd mod_wsgi mod_proxy git python-cherrypy python-jinja2`
+    - `git clone https://github.com/IIHE/dashboard.git`
+    - Copy the vhost-dashboard.conf to your Apache config directory (eg. /etc/http/conf.d/) and replace the following:
+      * AQ_URL - URL of your Aquilon server (including port)
```
backticks around the constants

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+    - If you want to hide aquilon related menu : Edit assets/js/pages/config.js and set AQ_URL to ''.
+
+* On the AII server
+    - `yum install perl-Config-Tiny`
```
add `-y`

stdweird replied:
but if this an AII-server, it will already be there since it's a dependency of CAF

samary replied:
You're right !
This was an old requirement for perl-Config-Simple

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+
+* On the AII server
+    - `yum install perl-Config-Tiny`
+    - `cd /var/www/cgi-bin && wget https://raw.githubusercontent.com/IIHE/dashboard/master/cgi-bin/dashboard-aii.cgi`
```
same remark: what is in the IIHE repo that can't be moved here?

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+    - Be sure your dashboard webserver can reach this page via HTTP GET requests
+
+* On the SVN client
+    - `cd /var/www/cgi-bin && wget https://raw.githubusercontent.com/IIHE/dashboard/master/cgi-bin/dashboard-svn.cgi`
```
same, `IIHE` url really needed?

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+
+* On the SVN client
+    - `cd /var/www/cgi-bin && wget https://raw.githubusercontent.com/IIHE/dashboard/master/cgi-bin/dashboard-svn.cgi`
+    - Edit dashboard-svn.cgi and set the SVN repository URL
```
backticks

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+* On the SVN client
+    - `cd /var/www/cgi-bin && wget https://raw.githubusercontent.com/IIHE/dashboard/master/cgi-bin/dashboard-svn.cgi`
+    - Edit dashboard-svn.cgi and set the SVN repository URL
+    - Check if the webservice (eg: apache) can run the `/usr/bin/svn` command via sudo (/etc/sudoers)
```
backticks around the apache; and apache is the user that runs teh service, not the service itself

samary replied:
@stdweird : `Apache` is the name of the product (http://en.wikipedia.org/wiki/Web_server#Market_share)
It is called `httpd` under RHEL but `apache2` under Debian
Also, you can run 'Apache' under another user. That's why I made this procedure as standard as possible (for nginx, lighttpd, ...)

stdweird commented on README.md:
```diff
@@ -0,0 +1,53 @@
+
+Licence & Credit
+================
+Project forked and extended from Aquilon Console : https://github.com/amazerfrazer/aquilonconsole
```
and the license?

jrha replied:
Should be Apache2 like everything else, `aquilonconsole` originally came from @njwilliams's work.

samary replied:
OK for our code, but need to gather licences for third-party tools (jquery, bootstrap, ...)

jrha replied:
Indeed you do :smile:

samary commented:
@stdweird I'm trying to use `CAF::Process` but I get Tainted issues : 

`Insecure dependency in exec while running with -T switch at /usr/lib/perl/LC/Process.pm line 475.`

With this code : 

```
my ($action, $hostname) = @_;

  $hostname =~ /^([\w\-\.]+)$/;
  $action =~ /^(configure|install|reinstall|boot)$/;

  $ENV{PATH}="/bin:/usr/bin:/sbin:/usr/bin:/usr/sbin";

  my @command = qw(/usr/bin/sudo /usr/sbin/aii-shellfe);

    if ($action eq 'reinstall') {
        push(@command, '--remove', '--configure', '--install');
    }
    else {
      push(@command, '--$action');
    }

    push(@command, $hostname);

     my $p = new CAF::Process(\@command);
     my $output = $p->output();

      print "Content-type: text/plain\n\n";

      # do something with $? exitcode in case of failure
      print "Failed to $action $hostname using aii-shellfe<br>" if ($? ne 0);
      print "$output<br>";
```

Do you know where I made a mistake ? I expected untainting lines working...  

UPDATE : 

I forget some `$hostname = $1` after untainting. Sorry for the noise

Commit:
Update README according to @stdweird remarks.

Commit:
use CAF::Process + stricter regexp + code style

Commit:
add newline at the end of the file

Commit:
Add third-party licenses

stdweird commented on cgi-bin/dashboard-svn.cgi:
```diff
@@ -0,0 +1,40 @@
+        print "Content-type: text/xml\n\n$output";
+    }
+    else {
+        print "Content-type: text/plain\n\nError while retrieving SVN Logs : $?!";
```
did you really mean `$?!` and not `$?` or `$!`?

stdweird replied:
nvm

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,353 @@
+my $profile_prefix = $cfg->{_}->{profile_prefix} eq 0 ? '' : $cfg->{_}->{profile_prefix};
+
+# cdb url
+my $cdb_url = $cfg->{_}->{cdb_url};
```
the parameter in the config files is without the `_`, so just `cdburl`

stdweird commented on cgi-bin/dashboard-aii.cgi:
```diff
@@ -0,0 +1,353 @@
+  my $hostname = @_;
+  if ($hostname =~ m/^([\w\-\.]+)$/) {
+
+    my $file = "$profiles_dir/$profile_prefix$hostname.json";
```
to support gzipped json, maybe you can do

``` perl
my $filename = "$profiles_dir/$profile_prefix$hostname.json";
my $fh;
if (-f $filename) {
    $fh =  new IO::String;
} elsif ( -f "$filename.gz") {
    $filename .= ".gz";
    $fh =  new IO::Zlib;
} else {
   die (no such file)
}

$fh->open($filename, 'rb');
my $json_text = join('', <$fh>);
$fh->close();
```

(very untested :smile: )

samary replied:
@stdweird : I got it working with IO::File

stdweird commented:
@samary I made some minor remarks, but one issue I have is security related. I think it should come with some disclaimer that this might expose sensitive data by default (e.g. some warning in the `README.md` that this page should be hosted behind some authentication).

Do you think we can get this released before end-of-month (together with the release)?

samary commented:
@stdweird I think can address your remarks for the release. In what form do you think this should be released (clone from github, tarball, scripts, RPM) ?

Commit:
Factorisation for profile reading
Same regexp for hostnames

Commit:
[FIX] cdburl typo

Commit:
Use cleaner arg handling

jrha commented:
RPM would be perfect, but we can live with imperfect for the first release I think.

samary commented:
@jrha I'm doing a complete review of this pull request. I'll let you know when I think it is ready for merge.

Commit:
Use cdburl to fetch profiles and work only with file:// protocol (for now)

Commit:
Format using Quattor style guidelines

Commit:
Cleaner die

Commit:
Cleanup variable and correct scopes

Commit:
Use profile_format from config

Commit:
Handle json.gz profiles
Needs to be defined in aii-shellfe conf file (profile_format=json.gz)
Add new requirement : IO::File and IO::Zlib

samary commented:
@stdweird How does AII handles the profile_format (not required) and the profile_prefix (0 seen as empty) from aii-shellfe.conf ? I cannot rely on these parameters if they are not consistent across diverse installation.

Commit:
[FIX] profile prefix misunderstanding

Commit:
[FIX] disable buttons

Commit:
Factorize common functions

Commit:
code cleanup

Commit:
[FIX] empty variable concatenation + use 0 for false

Commit:
rename cpu to cores

Commit:
Add hdcapacity
